homeassistant/components/asuswrt/config_flow.py

```diff
@@ -76,25 +74,30 @@ def _show_setup_form(self, user_input=None, errors=None):
         if user_input is None:
             user_input = {}
 
+        schema = {
+            vol.Required(CONF_HOST, default=user_input.get(CONF_HOST, "")): str,
+            vol.Required(CONF_USERNAME, default=user_input.get(CONF_USERNAME, "")): str,
+            vol.Optional(CONF_PASSWORD)
+            if self.show_advanced_options
+            else vol.Required(CONF_PASSWORD): str,
+            vol.Required(CONF_PROTOCOL, default=PROTOCOL_SSH): vol.In(
+                {PROTOCOL_SSH: "SSH", PROTOCOL_TELNET: "Telnet"}
+            ),
+        }
+
+        if self.show_advanced_options:
+            schema[vol.Required(CONF_PORT, default=0)] = vol.All(
+                vol.Coerce(int), vol.Range(min=0, max=65535)
+            )
+            schema[vol.Optional(CONF_SSH_KEY)] = str
+
+        schema[vol.Required(CONF_MODE, default=MODE_ROUTER)] = vol.In(
```

Comment:
Can't we detect this from the Asus config ?

Reply:
Not as I know. May be some information from `NVRAM`, but I don't know which and may be the library do not expose it.